pkg/services/alerting/notifiers/telegram.go

```diff
@@ -107,17 +104,99 @@ func (this *TelegramNotifier) Notify(evalContext *alerting.EvalContext) error {
 		message = message + fmt.Sprintf("\n<i>Metrics:</i>%s", metrics)
 	}
 
-	bodyJSON.Set("text", message)
+	var body bytes.Buffer
+	w := multipart.NewWriter(&body)
+	fw, _ := w.CreateFormField("chat_id")
+	fw.Write([]byte(this.ChatID))
+
+	fw, _ = w.CreateFormField("parse_mode")
+	fw.Write([]byte("html"))
 
-	url := fmt.Sprintf(telegeramApiUrl, this.BotToken, "sendMessage")
-	body, _ := bodyJSON.MarshalJSON()
+	fw, _ = w.CreateFormField("text")
+	fw.Write([]byte(message))
 
+	w.Close()
+
+	this.log.Info("Sending telegram text notification", "chat_id", this.ChatID, "bot_token", this.BotToken)
+	url := fmt.Sprintf(telegramApiUrl, this.BotToken, "sendMessage")
 	cmd := &m.SendWebhookSync{
 		Url:        url,
-		Body:       string(body),
+		Body:       body.String(),
 		HttpMethod: "POST",
+		HttpHeader: map[string]string{
+			"Content-Type": w.FormDataContentType(),
+		},
 	}
 
+	return cmd
+}
+
+func (this *TelegramNotifier) buildImageMessage(evalContext *alerting.EvalContext) *m.SendWebhookSync {
+	message := fmt.Sprintf("%s\nState: %s\nMessage: %s\n", evalContext.GetNotificationTitle(), evalContext.Rule.Name, evalContext.Rule.Message)
+
+	ruleUrl, err := evalContext.GetRuleUrl()
+	if err == nil {
+		message = message + fmt.Sprintf("URL: %s\n", ruleUrl)
+	}
+
+	metrics := ""
+	fieldLimitCount := 4
+	for index, evt := range evalContext.EvalMatches {
+		metrics += fmt.Sprintf("\n%s: %s", evt.Metric, evt.Value)
+		if index > fieldLimitCount {
+			break
+		}
+	}
+	if metrics != "" {
+		message = message + fmt.Sprintf("\nMetrics:%s", metrics)
+	}
+
+	f, err := os.Open(evalContext.ImageOnDiskPath)
+	if err != nil {
+		return nil // fall back to text message
+	}
+	defer f.Close()
+
+	var body bytes.Buffer
+
+	w := multipart.NewWriter(&body)
+	fw, _ := w.CreateFormField("chat_id")
+	fw.Write([]byte(this.ChatID))
+
+	fw, _ = w.CreateFormField("caption")
+	fw.Write([]byte(message))
+
+	fw, _ = w.CreateFormFile("photo", evalContext.ImageOnDiskPath)
+	io.Copy(fw, f)
+
+	w.Close()
+	this.log.Info("Sending telegram image notification", "photo", evalContext.ImageOnDiskPath, "chat_id", this.ChatID, "bot_token", this.BotToken)
+
+	url := fmt.Sprintf(telegramApiUrl, this.BotToken, "sendPhoto")
+	cmd := &m.SendWebhookSync{
+		Url:        url,
+		Body:       body.String(),
+		HttpMethod: "POST",
+		HttpHeader: map[string]string{
+			"Content-Type": w.FormDataContentType(),
+		},
+	}
+
+	return cmd
+}
+
+func (this *TelegramNotifier) Notify(evalContext *alerting.EvalContext) error {
+	var cmd *m.SendWebhookSync
+	if evalContext.ImagePublicUrl == "" && this.UploadImage == true {
+		_, err := os.Stat(evalContext.ImageOnDiskPath)
+		if err == nil {
+			cmd = this.buildImageMessage(evalContext)
+		}
+	}
+
+	if cmd == nil {
```

Comment:
This should be an else statement for `evalContext.ImagePublicUrl == "" && this.UploadImage == true`